packages/twenty-front/src/modules/auth/sign-in-up/hooks/useWorkspaceBypass.ts

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import { useRecoilState, useRecoilValue } from 'recoil';
+
+import { isMultiWorkspaceEnabledState } from '@/client-config/states/isMultiWorkspaceEnabledState';
+import { useIsCurrentLocationOnAWorkspace } from '@/domain-manager/hooks/useIsCurrentLocationOnAWorkspace';
+import { workspaceAuthBypassProvidersState } from '@/workspace/states/workspaceAuthBypassProvidersState';
+import { workspaceAuthProvidersState } from '@/workspace/states/workspaceAuthProvidersState';
+import { workspaceBypassModeState } from '@/workspace/states/workspaceBypassModeState';
+
+export const useWorkspaceBypass = () => {
+  const workspaceAuthProviders = useRecoilValue(workspaceAuthProvidersState);
+  const workspaceAuthBypassProviders = useRecoilValue(
+    workspaceAuthBypassProvidersState,
+  );
+  const [workspaceBypassMode, setWorkspaceBypassMode] = useRecoilState(
+    workspaceBypassModeState,
+  );
+
+  const { isOnAWorkspace } = useIsCurrentLocationOnAWorkspace();
+  const isMultiWorkspaceEnabled = useRecoilValue(isMultiWorkspaceEnabledState);
+
+  const isWorkspaceContext =
+    isOnAWorkspace || isMultiWorkspaceEnabled === false;
+
+  const hasOnlySSOProvidersEnabled = useMemo(() => {
+    if (!workspaceAuthProviders) {
+      return false;
+    }
+
+    return (
+      workspaceAuthProviders.sso.length > 0 &&
+      !workspaceAuthProviders.google &&
+      !workspaceAuthProviders.microsoft &&
+      !workspaceAuthProviders.password
+    );
+  }, [workspaceAuthProviders]);
+
+  const hasBypassProvidersAvailable = useMemo(() => {
+    if (!workspaceAuthBypassProviders) {
+      return false;
+    }
+
+    const { google, microsoft, password } = workspaceAuthBypassProviders;
+
+    return google || microsoft || password;
+  }, [workspaceAuthBypassProviders]);
+
+  const shouldOfferBypass =
+    isWorkspaceContext &&
+    hasOnlySSOProvidersEnabled &&
+    hasBypassProvidersAvailable;
+
+  useEffect(() => {
```

Comment:
We only allow effects in dedicated Effect components or in a very specific cases ; in most case case you shouldn't even need the effect component

https://react.dev/learn/you-might-not-need-an-effect